workspaces/kubernetes-native-dxp/k8s/custom-rest-service/ingress.yaml

```diff
@@ -0,0 +1,17 @@
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: custom-rest-service-localhost
```

Comment:
Having one ingress per service causes problems. With a single ingress having rules for each domain we get the expected behaviour. I'm already using this with TLS enabled locally and it's working. We need a couple configurations for DXP to render protocols as `https`, but that's expected. All I have to do is add the domains to my hosts file:

I have this broken out into a separate `ingress.sh` which loads at the end of `run_local.sh`:
```bash
#!/bin/bash

kubectl delete ingress common-ingress --ignore-not-found=true
kubectl create ingress common-ingress --class=nginx \
	--rule="dxp.localdev.me/*=dxp-service:80,tls" \
	--rule="custom-rest-service.localdev.me/*=custom-rest-service:80,tls" \
	--rule="remote-app-a.localdev.me/*=remote-app-a:80,tls"
```

Reply:
I'm fine with this simplification.  However, the next step I was going to push (if we liked this) was to move to "helm" for all of the "extensions".   Because there is so much bolierplate for these extensions, a helm chart for existing (and new ) extensions would be natural.  I was even thinking in the demo, we could add a "new" extension bey just adding a directory and a single line in a helm chart.  But Maybe we don't want to push this too far, since the user's dev enviornment is going to look drasitically different in the end.

Reply:
yeah, I'd like that idea of helm. I also like the idea of demonstrating adding a new service. I was thinking we could do that using the cronjob scenario where we add a cronjob that does headless using the "Headless Server" application profile.

Reply:
I'll update skaffold to use "common" ingress instead of "per service".

Reply:
Sent a new push with these changes.